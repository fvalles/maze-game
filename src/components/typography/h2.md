Basic h2 text component with default color="text":

```jsx
<H2>Lorem ipsum</H2>
```

H2 text component with color="white":

```jsx
<H2 color="white">Lorem ipsum</H2>
```
