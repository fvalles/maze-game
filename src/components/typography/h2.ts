import styled from 'styled-components';
import { TypographyProps } from './common';

export const H2 = styled.h2<TypographyProps>`
  color: ${({ color = 'text', theme }) => theme.Colors[color]};
  font-family: ${({ theme }) => theme.Fonts.light}, 'Avenir Next', 'Segoe UI',
    sans-serif;
  font-size: 40px;
  line-height: 44px;
`;
