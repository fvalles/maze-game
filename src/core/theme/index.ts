export { Fonts } from './fonts';
export { GlobalStyle } from './global-style';
