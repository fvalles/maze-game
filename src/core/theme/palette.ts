export const palette = {
  // color names are based on https://colornamer.netlify.app/
  artfulRed: '#B20600',
  crystalCut: '#F9F2ED',
  blackMarket: '#212240',
  janitor: '#2666CF',
  white: '#FFFFFF',
};
