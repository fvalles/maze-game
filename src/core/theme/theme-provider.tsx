import React, { FunctionComponent } from 'react';
import { DefaultTheme, ThemeProvider as Provider } from 'styled-components';
import { Fonts } from '.';

interface ThemeProviderProps {
  children?: React.ReactNode;
}

export const ThemeProvider: FunctionComponent<ThemeProviderProps> = ({
  children,
}) => (
  <Provider
    theme={
      {
        Fonts,
      } as DefaultTheme
    }>
    {children}
  </Provider>
);
