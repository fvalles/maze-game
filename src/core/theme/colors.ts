import { palette } from './palette';

export const Colors = {
  // color names are based on https://colornamer.netlify.app/
  buttonBackground: palette.artfulRed,
  modalBackground: palette.crystalCut,
  restartButtonBackground: palette.janitor,
  text: palette.blackMarket,
  white: palette.white,
};

export type ColorsType = typeof Colors;
export type KeyColors = keyof ColorsType;
