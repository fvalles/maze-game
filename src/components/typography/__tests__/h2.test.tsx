import React from 'react';
import { ThemeProvider } from '@core/theme/theme-provider';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { H2 } from '../h2';

describe('h2 component', () => {
  it('should render with default css properties', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <H2>Lorem ipsum</H2>
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toHaveStyleRule('color', '#212240');
    expect(result).toHaveStyleRule(
      'font-family',
      "Avenir Light,'Avenir Next','Segoe UI', sans-serif",
    );
    expect(result).toHaveStyleRule('font-size', '40px');
    expect(result).toHaveStyleRule('line-height', '44px');
    expect(result).toMatchSnapshot();
  });

  it('should render with custom white color', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <H2 color="white">Lorem ipsum</H2>
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toHaveStyleRule('color', '#FFFFFF');
    expect(result).toHaveStyleRule(
      'font-family',
      "Avenir Light,'Avenir Next','Segoe UI', sans-serif",
    );
    expect(result).toHaveStyleRule('font-size', '40px');
    expect(result).toHaveStyleRule('line-height', '44px');
    expect(result).toMatchSnapshot();
  });
});
