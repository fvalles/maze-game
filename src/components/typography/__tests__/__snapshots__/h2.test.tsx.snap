// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`h2 component should render with custom white color 1`] = `
.c0 {
  color: #FFFFFF;
  font-family: Avenir Light,'Avenir Next','Segoe UI', sans-serif;
  font-size: 40px;
  line-height: 44px;
}

<h2
  className="c0"
  color="white"
>
  Lorem ipsum
</h2>
`;

exports[`h2 component should render with default css properties 1`] = `
.c0 {
  color: #212240;
  font-family: Avenir Light,'Avenir Next','Segoe UI', sans-serif;
  font-size: 40px;
  line-height: 44px;
}

<h2
  className="c0"
>
  Lorem ipsum
</h2>
`;
